### PR TITLE
Use latest prometheus lib to construct roundtripper

### LINKED ISSUE #39: Bump dep version for "github.com/prometheus/common"
The code below

https://github.com/paypal/load-watcher/blob/1c92ae3f9346943cb3694e7de24c95f851508b61/pkg/watcher/internal/metricsprovider/prometheus.go#L66

is no longer valid in the latest [Prometheus library](https://github.com/prometheus/common/blob/main/config/http_config.go).

The code should be refactored to:

```go
config.NewAuthorizationCredentialsRoundTripper("Bearer", token, rt)
```

This is found when I wanted to bump the version of scheduler-plugins to k8s 1.22.

### PR DESCRIPTION
**Pull request by Huang-Wei**
Fixes #39.

The upgrade of scheduler-plugins (to 1.22) is blocked on this.

/assign @zorro786 
/cc @wangchen615

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,8 +4,8 @@ go 1.15
 
 require (
 	github.com/francoispqt/gojay v1.2.13
-	github.com/prometheus/client_golang v1.9.0
-	github.com/prometheus/common v0.15.0
+	github.com/prometheus/client_golang v1.11.0
+	github.com/prometheus/common v0.26.0
 	github.com/sirupsen/logrus v1.6.0
 	github.com/stretchr/testify v1.5.1
 	k8s.io/apimachinery v0.19.0
```
Comment (Huang-Wei) [inline go.mod:11]: I'm not sure if you'd like to bump this to v0.22.x as well.

Comment (zorro786) [reply go.mod:11]: @Huang-Wei Please update this one too

Comment (Huang-Wei) [reply go.mod:11]: ok, k8s v1.22.3 should be good, which vendors the following:

```
	github.com/prometheus/client_golang v1.11.0
	github.com/prometheus/client_model v0.2.0
	github.com/prometheus/common v0.26.0
```

**Review by zorro786 — APPROVED** [review]
LGTM. Thanks @Huang-Wei for raising this PR.

**Comment by zorro786** [comment]
@Huang-Wei I will release v0.2.1 along with #38 tomorrow. So that you can use this tag in `master` branch (or any new cut off) of scheduler-plugins.

**Comment by Huang-Wei** [comment]
@zorro786 Thanks!

**Review by wangchen615 — APPROVED** [review]

**Comment by Huang-Wei** [comment]
It's approved but somehow not merged?

**Pull request merged**